File: frontend/src/models/FoodSpot.ts
import {Foodlocation} from "./Foodlocation";
import {FoodType, Place, Space} from "./FoodData";

export type FoodSpot = {
    id: string;
    name: string;
    placeType: Place[];
    spaceType: Space[];
    allergens: FoodType[];
    position: Foodlocation
}

File: backend/src/main/java/de/neuefische/backend/models/FoodSpot.java
package de.neuefische.backend.models;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Document("foodspot")
public record FoodSpot(@Id String id,
                       String name,
                       List<String> spaceType,
                       List<String> allergens,
                       String placeType,
                       Foodlocation position) {
}
